pypeit/reduce.py

```diff
@@ -390,13 +392,23 @@ def run(self, basename=None, ra=None, dec=None, obstime=None,
                 self.global_sky = self.initial_sky.copy()
             else:
                 self.global_sky = self.global_skysub(skymask=self.skymask, show=self.reduce_show)
+
+            # Apply a global flexure correction to each slit
+            # provided it's not a standard star
+            if self.par['flexure']['spec_method'] != 'skip' and not self.std_redux:
+                self.spec_flexure_correct(basename+"_global", mode='slit', update_waveimg=True)
+
             # Extract + Return
             self.skymodel, self.objmodel, self.ivarmodel, self.outmask, self.sobjs \
                 = self.extract(self.global_sky, self.sobjs_obj)
             if self.ir_redux:
                 self.sobjs.make_neg_pos() if return_negative else self.sobjs.purge_neg()
 
         else:  # No objects, pass back what we have
+            # Apply a global flexure correction to each slit
+            # provided it's not a standard star
+            if self.par['flexure']['spec_method'] != 'skip' and not self.std_redux:
+                self.spec_flexure_correct(basename+"_global", mode='slit', update_waveimg=True)
             #Could have negative objects but no positive objects so purge them
             if self.ir_redux:
                 self.sobjs_obj.make_neg_pos() if return_negative else self.sobjs_obj.purge_neg()
```

Comment:
The line

 if self.par['flexure']['spec_method'] != 'skip' and not self.std_redux:
                 self.spec_flexure_correct(basename+"_global", mode='slit', update_waveimg=True)

Should be moved out of this if self.nobj> 0 else sequence, since it needs to be done either way. As it is coded here it is redundantly replicating the same line of code.

Reply:
Yeah, you are right. I agree. The logic is slightly tricky here, but I don't think it can be simplified.